### PR TITLE
Format with black and markdownlint

### PR DESCRIPTION
**Pull request by elliotwutingfeng**
## Changes

- Formatted **.py** files with [black](https://pypi.org/project/black)
- Formatted **README.md** with [markdownlint](https://github.com/DavidAnson/markdownlint)

### CODE REVIEW

**Comment by daxpedda** [comment]
Would you mind adding this to the CI?

**Comment by elliotwutingfeng** [comment]
Should the CI format and commit the changes or fail and leave it to the user to fix it?

**Comment by daxpedda** [comment]
Definitely not commit, just fail.
Could you also split it into different jobs? Both the Markdown and Python formatting.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,3 +12,28 @@ jobs:
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Format Markdown with markdownlint
+        run: |
+          npm install -g markdownlint-cli
+          markdownlint --disable MD013 --fix .
```
Comment (daxpedda) [inline .github/workflows/ci.yml:26]: I did check this out but couldn't really find a command that only checks.
`--fix` doesn't do what we want here, it just fixes things but doesn't return a failed status code if there was something to fix.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -12,3 +12,30 @@ jobs:
...
+          git add -A
+          git diff --name-only --cached --exit-code
```
Comment (daxpedda) [inline .github/workflows/ci.yml:28]: Why `--name-only`?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (elliotwutingfeng) [reply .github/workflows/ci.yml:26]: I've added `git diff` [`--exit-code`](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---exit-code) which should return 0 when there is nothing to fix, and return 1 when there is.

Comment (daxpedda) [reply .github/workflows/ci.yml:26]: Thank you!

**Comment by daxpedda** [comment]
Thank you! This was very useful and well done.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (elliotwutingfeng) [reply .github/workflows/ci.yml]: Oops, removing it now to include the error diff in the output.

Comment (daxpedda) [reply .github/workflows/ci.yml]: Already done :wink:.

**Pull request merged**